### PR TITLE
Fix Issue 20984 - Heisenbug: FreeBSD64 Phobos random test suite failu…

### PR DESCRIPTION
**Pull request by CyberShadow**
Replace the arbitrary wait duration with a poll, which should only succeed when the result is known to be ready.

### CODE REVIEW

**Comment by CyberShadow** [comment]
Oh the irony:

```
/home/braddr/sandbox/at-client/pull-4135049-FreeBSD_64_64/dmd/generated/freebsd/release/64/dmd -conf= -m64 -fPIC -g -lowmem -i -Itools -version=NoMain -unittest -run tools/d_do_test.d
gmake[1]: *** [Makefile:217: generated/d_do_test] Segmentation fault (core dumped)
```
Is that another test suite Heisenbug? :)

**Comment by MoonlightSentinel** [comment]
Looks like a recent regression, it started a few days ago and is probably related to/the same issue as the `unit` segfault.

File: std/process.d [hunk 1]
```diff
@@ -1609,50 +1609,58 @@ version (Posix) @system unittest
...
+            wait(pid);
         assert(readText(patho).chomp() == "INPUT output bar");
         assert(readText(pathe).chomp().stripRight() == "INPUT error baz");
-        remove(pathi);
-        remove(patho);
-        remove(pathe);
+        while (remove(pathi).collectException) Thread.sleep(10.msecs);
+        while (remove(patho).collectException) Thread.sleep(10.msecs);
+        while (remove(pathe).collectException) Thread.sleep(10.msecs);
+        while (remove(done).collectException) Thread.sleep(10.msecs);
```
Comment (n8sh) [inline std/process.d:1663]: I'd suggest not including this change unless it is solving a known problem.

Comment (CyberShadow) [reply std/process.d:1663]: Yes, they're solving a problem. Without this change there was another random failure. The files were still open by the shell even though the content was written to it.

**Pull request merged**